src/components/Header/NavComponents/userprofile.tsx

```diff
@@ -57,18 +127,96 @@ export const UserProfile: React.FC<{}> = () => {
         bg={mode({ base: "secondary.800", md: "primary.500" }, "primary.500")}
         rounded="lg"
         px="5px"
+        mr="2px"
       >
         <Text mr="5px">{userBal}</Text>
         <Text fontWeight="400">AGVE</Text>
       </Center>
+      {connector !== "" ? (
+        connector === "metamask" ? (
+          <Popover>
+            <PopoverTrigger>
+              <Button
+                minWidth="9rem"
+                height={{ base: "4rem", md: "3rem" }}
+                fontSize={{ base: "4xl", md: "2xl" }}
+                mx="1.5rem"
+                px="1.5rem"
+                pt={{ base: "0px", md: "4px" }}
+                color="white"
+                bg={mode(
+                  { base: "secondary.800", md: "primary.500" },
+                  "primary.500"
+                )}
+                rounded="lg"
+              >
+                {error && error instanceof UnsupportedChainIdError ? (
+                  <Text fontWeight="400">Invalid Chain</Text>
+                ) : (
+                  <Text fontWeight="400">Chain: {currChainName}</Text>
+                )}
+              </Button>
+            </PopoverTrigger>
+            <PopoverContent
+              bg={mode(
+                { base: "primary.900", md: "primary.900" },
+                "primary.900"
+              )}
+              color="white"
+              borderColor={mode(
+                { base: "primary.50", md: "primary.50" },
+                "primary.50"
+              )}
+            >
+              <PopoverArrow
+                bg="#007c6e"
+                color="white"
+                border="0px"
+                boxShadow="-1px -1px 1px 0px #36cfa2 !important"
+                borderColor={mode(
+                  { base: "primary.50", md: "primary.50" },
+                  "primary.50"
+                )}
+              />
+              <PopoverBody>
+                <Stack spacing={4} direction="column">
+                  {popoverData}
+                </Stack>
+              </PopoverBody>
+            </PopoverContent>
+          </Popover>
+        ) : (
+          <Center
+            minWidth="12rem"
+            height={{ base: "4rem", md: "3rem" }}
+            fontSize={{ base: "4xl", md: "2xl" }}
+            mx="1.5rem"
+            px="1rem"
+            color="white"
+            bg={mode(
+              { base: "secondary.800", md: "primary.500" },
+              "primary.500"
+            )}
+            rounded="lg"
+          >
+            {error && error instanceof UnsupportedChainIdError ? (
+              <Text fontWeight="400">Invalid Chain</Text>
+            ) : (
+              <Text fontWeight="400">Chain: {currChainName}</Text>
+            )}
+          </Center>
+        )
+      ) : (
+        <></>
+      )}
```

Comment:
Each of these logical branches should be a separate component, at this scale.

Reply:
Done